src/main/java/ReviewGateways/ReviewGateway.java

```diff
@@ -0,0 +1,270 @@
+        }
+        else{
+            return ID;
+        }
+    }
+
+    /*
+    After using the current ID to make a new review, increase it by 1
+     */
+    @Override
+    public void incrementReviewID() throws IOException {
+        //Get current ID and increment it by 1
+        int currentID = Integer.parseInt(this.loadReviewID());
+        currentID = currentID + 1;
+        String stringID = Integer.toString(currentID);
+
+        //Try to write this to the database
+        File file = new File(NAME_OF_REVIEW_ID_COUNTER);
+        FileWriter writer = new FileWriter(file);
+        writer.write(stringID);
+        writer.close();
+    }
+
+    /*
+    Method to retrieve a single Review using its ID
+     */
+    @Override
+    public Review getReview(String id) throws FileNotFoundException, ReviewNotFoundException {
+        //Attach a scanner to the database and declare a variable to hold one line from the database
+        File file = new File(NAME_OF_REVIEW_DATABASE);
+        Scanner scanner = new Scanner(file);
+        String line;
+
+        //While there are lines in the database
+        while(scanner.hasNext()){
+            //Get each line and split it into pieces by the delimiter
+            line = scanner.nextLine();
+            String[] pieces = line.split(DELIMITER);
+
+            //If the piece where ID is stored equals the provided ID, then return a Review made from this line
+            if(pieces[ID_INDEX].equals(id)){
+                return makeReviewObject(pieces);
+            }
+
+        }
+        //No reviews matching this ID are found, so throw ReviewNotFoundException
+        throw new ReviewNotFoundException("No reviews were found matching that id.");
+    }
+
+    /*
+    Return a list of Reviews based on a list of IDs. Exact same as retrieving a single review, except we iterate until
+    ids.length reviews are added to the list of Reviews to return, or we run out of lines
+     */
+    @Override
+    public ArrayList<Review> getReviews(ArrayList<String> ids) throws FileNotFoundException {
+        ArrayList<Review> reviews = new ArrayList<>();
+        int length = ids.size();
+        int counter = 0;
+        File file = new File(NAME_OF_REVIEW_DATABASE);
+        Scanner scanner = new Scanner(file);
+        String line;
+
+        while(scanner.hasNext()){
+            line = scanner.nextLine();
+            String[] pieces = line.split(DELIMITER);
+
+            if(ids.contains(pieces[ID_INDEX])){
+                Review review = makeReviewObject(pieces);
+                reviews.add(review);
+                counter = counter + 1;
+
+                if(counter == length){
+                    return reviews;
+                }
+            }
+
+        }
+        return reviews;
+    }
+
+    /*
+    Add a Review to the database using the nine attributes
+     */
+    @Override
+    public void addReview(String id, String stars, String text, String username, String restaurantLocation,
```

Comment:
Is it possible to move all arguments inside of the body of the method? so instead of passing bunch of xxx.getxxx, just pass the object, which simplifies the method call.

Reply:
Will do